### DISCUSSION TITLE
Where are models installed by default ?

### DISCUSSION THREAD

**Discussion by Infinitusvoid** [huggingface/diffusers#6675] [Q&A] [answered]
When you run this code where is the model located how to find out the path to the model itself ? 
``
from diffusers import StableDiffusionPipeline
import torch

model_id = "stabilityai/stable-diffusion-2-1"
pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
``

**Comment by a-r-r-o-w** [comment, marked as answer]
Hi @Infinitusvoid.

diffusers uses `huggingface/huggingface_hub` to download and store models that you use. [This](https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables) provides you with a list of environment variables that hub uses by default.

`HF_HUB_CACHE` is what you're looking for, and by default it is `~/.cache/huggingface/hub`, where `~` means your home directory for your profile on windows/linux/mac.

If you'd like to change the location where your models are downloaded by default, you could do this:
  ```bash
  # If you want the variable to persist everytime you run a shell
  echo "export HF_HOME=/home/profile/path/to/model/directory" >> ~/.bashrc # or config file for whatever terminal you use

  # You could also create a symbolic link from your directory to HF_HUB_CACHE
  ln -s /path/to/your/model/directory $HF_HUB_CACHE
  ```

You could also achieve the same by using the [`cache_dir`](https://huggingface.co/docs/diffusers/v0.8.0/en/using-diffusers/loading#diffusers.ModelMixin.from_pretrained.cache_dir) parameter when downloading models:

```py
model_id = "stabilityai/stable-diffusion-2-1"
pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16, cache_dir="/path/to/model/directory")
```

**Reply by Infinitusvoid** [reply]
Thanks, found 70 Gb of models saved there xD

**Reply by runner22k** [reply]
does diffuser has a config file? i am on windows 11 and i want to change the default models location to a AI dedicated drive. Please help.

**Reply by TILLY9000** [reply]
@a-r-r-o-w  I’d like to delete my downloaded models as well, but don’t have a ~/.cache folder on my mac. Any idea where to find them in this case?

**Reply by a-r-r-o-w** [reply]
What output do you get when you run `echo $HF_HOME` in your terminal?

**Comment by TILLY9000** [comment]
I get an empty line.

**Reply by a-r-r-o-w** [reply]
I'm not too familiar with Mac so I can only provide a rough solution on what to do. Could you try running `du -h -d 1 /` in your terminal? This should tell you the size of each directory in your root folder. You can then see what directories are using the highest size, go to that directory, and repeat.

As such, all downloaded models _should_ be available in `~/.cache/huggingface/` if you haven't made any custom modifications to the save path
